File: mek_stripe_terminal/android/src/main/kotlin/mek/stripeterminal/mappings/TipMappings.kt
package mek.stripeterminal.mappings

import com.stripe.stripeterminal.external.models.Tip
import com.stripe.stripeterminal.external.models.TippingConfiguration
import TipApi
import TippingConfigurationApi

fun Tip.toApi(): TipApi {
    return TipApi(
        amount = amount
    )
}

// PARAMS

fun TippingConfigurationApi.toHost(): TippingConfiguration {
    return TippingConfiguration.Builder().setEligibleAmount(eligibleAmount).build()
}

File: mek_stripe_terminal/ios/mek_stripe_terminal/Sources/mek_stripe_terminal/Mappings/TipMappings.swift
import Foundation
import StripeTerminal

extension Tip {
    func toApi() -> TipApi {
        return TipApi(
            amount: amount?.toInt64()
        )
    }
}

// PARAMS

extension TippingConfigurationApi {
    func toHost() throws -> TippingConfiguration {
        return try TippingConfigurationBuilder()
            .setEligibleAmount(eligibleAmount.toInt())
            .build()
    }
}
